test/test_nn.py

```diff
@@ -3459,6 +3459,20 @@ def test_Conv2d_deterministic_cudnn(self, dtype=torch.float):
             self.assertEqual(conv1.bias.grad.data, conv2.bias.grad.data, prec=0.0)
             self.assertEqual(conv1.weight.grad.data, conv2.weight.grad.data, prec=0.0)
 
+    @unittest.skipIf(not TEST_CUDNN, 'CUDNN not available')
+    def test_errorMsg_on_determinstic_and_benchmark_flag(self):
+        def fn():
+            inputs = torch.randn(2, 3, 5, 5, requires_grad=True)
+            with cudnn.flags(enabled=True, benchmark=True, deterministic=True):
```

Comment:
I think an easier and warning-generating test would be:
```
with cudnn.flags(enabled=True):
    cudnn.benchmark = True
    cudnn.deterministic = True
```
and
```
with cudnn.flags(enabled=True):
    cudnn.deterministic = True
    cudnn.benchmark = True
```

Reply:
yeah, I tried to set global flags, but it gives me this error:
```
RuntimeError: not allowed to set torch.backends.cudnn flags after disable_global_flags; please use flags() context manager instead
```
I think the global flags are guarded by https://github.com/pytorch/pytorch/blob/87b2f05a9c7750162fd4586e9c5154a64532eae3/torch/backends/cudnn/__init__.py#L151-L159